WordPress/Classes/ViewRelated/Blog/BlogListViewController.m

```diff
@@ -590,11 +541,73 @@ - (void)setEditing:(BOOL)editing animated:(BOOL)animated
     }
 }
 
+- (void)toggleRightBarButtonItems:(BOOL)enabled
+{
+    for (UIBarButtonItem *buttonItem in self.navigationItem.rightBarButtonItems) {
+        buttonItem.enabled = enabled;
+    }
+}
+
 - (void)toggleSearch
 {
     self.searchController.active = !self.searchController.active;
 }
 
+- (void)addSite
+{
+    UIAlertController *addSiteAlertController = [UIAlertController alertControllerWithTitle:nil
+                                                                                    message:nil
+                                                                             preferredStyle:UIAlertControllerStyleActionSheet];
+    UIAlertAction *addNewWordPressAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Create WordPress.com site", @"Create WordPress.com site button")
+                                                                    style:UIAlertActionStyleDefault
+                                                                  handler:^(UIAlertAction *action) {
+                                                                      [self showAddNewWordPressController];
+                                                                  }];
+    UIAlertAction *addSiteAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Add self-hosted site", @"Add self-hosted site button")
+                                                            style:UIAlertActionStyleDefault
+                                                          handler:^(UIAlertAction *action) {
+                                                              [self showLoginControllerForAddingSelfHostedSite];
+                                                          }];
+    UIAlertAction *cancel = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", @"Cancel button")
+                                                     style:UIAlertActionStyleCancel
+                                                   handler:nil];
+    
+    [addSiteAlertController addAction:addNewWordPressAction];
+    [addSiteAlertController addAction:addSiteAction];
+    [addSiteAlertController addAction:cancel];
+    addSiteAlertController.popoverPresentationController.barButtonItem = self.navigationItem.rightBarButtonItem;
```

Comment:
@kwonye nice, didn't see the docs for that. However, I would still think it might be preferred to have a weak reference to the specific button item. Otherwise if the order is changed for the button items at a later point in `configureNavigationBar`, it is not obvious that there is implementation relying on this ordering, even if only for UX purposes.